OpenRA.Mods.Common/Traits/Render/ProductionBar.cs

```diff
@@ -40,34 +41,44 @@ public ProductionBar(Actor self, ProductionBarInfo info)
 			this.info = info;
 		}
 
+		void INotifyCreated.Created(Actor self)
+		{
+			produces = self.TraitsImplementing<Production>().ToArray();
+			FindQueue();
+		}
+
 		void FindQueue()
 		{
-			var type = info.ProductionType ?? self.Info.TraitInfo<ProductionInfo>().Produces.First();
+			// Use all queues when no specific one is given
```

Comment:
Related to my other nit: wouldn't it make more sense to just FieldLoader.Require the queue type explicitly?

Reply:
Maybe; that sounds like a good idea to simplify the code here. Will have a look.